src/app.js

```diff
@@ -2,26 +2,70 @@
 //We've got some basic info about Karen's home
 //Debug the type of data provided
 //Return the types concatenated in a single variable
-
+function moreAboutHome(address, distanceFromTown, hasNeighbours) {
+    return typeof(address) +  typeof(distanceFromTown) +  typeof(hasNeighbours);
+    
+}
 //Progression 2:
 //Check if the data given is of the right type
 //parents = String, noOfSiblings = Number, isNuclearFamily = Boolean
+function moreAboutKaren(str, num, bool) {
+    if (typeof(str) === 'string' && typeof(num) === 'number' && typeof(bool) === 'boolean') {
+        return true;
+    }
+    return false;
+}
 
 //Progression 3:
 //Lily is suspicious about Karen's new friend
 //Karen tells her friend's age and even writes it down
 //Check which one those is not a number (NaN) and return that value
+function doesFriendExist(karenAge, friendAge) {
+    switch(true){
+        case isNaN(karenAge):
+            return karenAge;
+        case isNaN(friendAge):
+            return friendAge;
+        default:
+            return "Both ages are numbers";
+    }
+}
+
 
 //Progression 4:
 //Lily gave Karen x sweets
 //Karen ate y sweets herself
 //On her way to the river, she ate another z sweets every n meters travelled
 //Her friend divided the remaining sweets into 2 parts for each
 //How many sweets did her friend get to eat?
+function sweetTooth(sweets,riverDistance,friendSpeed){
```

Comment:
Reduce the line of codes. Think twice before implement.

Reply:
ok